src/lib/checkpoint/cp_mgr.cpp

```diff
@@ -218,20 +224,18 @@ void CPManager::cp_start_flush(CP* cp) {
     std::vector< folly::Future< bool > > futs;
     HS_PERIODIC_LOG(INFO, cp, "Starting CP {} flush", cp->id());
     cp->m_cp_status = cp_status_t::cp_flushing;
-
     for (size_t svcid = 0; svcid < (size_t)cp_consumer_t::SENTINEL; svcid++) {
-        if (svcid == (size_t)cp_consumer_t::REPLICATION_SVC) {
-            continue;
-        }
+        if (svcid == (size_t)cp_consumer_t::SEALER) { continue; }
         auto& consumer = m_cp_cb_table[svcid];
         if (consumer) { futs.emplace_back(std::move(consumer->cp_flush(cp))); }
     }
 
     folly::collectAllUnsafe(futs).thenValue([this, cp](auto) {
-        // Sync flushing replication svc at last as the cp_lsn updated here
-        // other component should at least flushed to cp_lsn
-        auto& repl_cp = m_cp_cb_table[(size_t)cp_consumer_t::REPLICATION_SVC];
-        if (repl_cp) {repl_cp->cp_flush(cp).wait();}
+        // Sync flushing SEALER svc which is the replication service
```

Comment:
sorry, I am still confused here. why SEALER here stand for replication service. maybe I miss something?

Reply:
we make SEALER = REPLICATION_SVC = 3.

There is one special service , in our case , is replication svc , act as a sealer ,  it "seals" a cp by labling which LSN the cp contains up to. Also , it needs to be flushed at last.  So whatever we read out from the *sealer* svc ,  it guarantee other svcs also at least flushing to this point

Reply:
oh, I see, actually SEALER is an alias of REPLICATION_SVC, if we have another svc need to be flush at last, then it will be the new sealer, right?

Reply:
We dont support 2 svcs need to be flushed at last ATM. Will extend based on need.